### PR TITLE
DPL-753-2 process new QC fields from TOL

### PR DESCRIPTION
**Pull request by Sangeetha-Bheeman**
Closes #

Changes proposed in this pull request:

- [ ] In tol_lab_share project, add new message properties for each of the new qc fields and add them as properties for a sample with no validation apart from checking is a string (check how it is with CommonName) (in message_properties/sample.py)

- [ ]  Create new message for qc_results traction to include the new fields in the payload we send

 

- [ ] Add the new properties into the qc results traction message that we will send (in message_properties/labware.py). We need also to add the attributes to qc results traction message interface to add the new fields

 

- [ ] Modify the CreateLabwareProcessor to send the message to qc_results traction endpoint if the creation of samples was successful in previous step

 

- [ ] In case of errror while sending qc_result request, append the errors to the feedback message

 

- [ ] Add the required tests.

*
*
* ...

### CODE REVIEW

File: tol_lab_share/messages/traction_qc_message.py [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+        will create a new instance and return it.
+        Parameters:
+        position (int) position that we want to return
+        Returns:
+        OutputTractionMessageRequest with the request required
+        """
+        if position not in self._requests:
+            self._requests[position] = TractionQcMessageRequest()
+
+        return self._requests[position]
```
Comment (emrojo) [inline tol_lab_share/messages/traction_qc_message.py:240]: Is this requests from the new qc result message sent to traction?

**Review by emrojo — APPROVED** [review]
Some minor comments, but all ok

File: tol_lab_share/config/defaults.py [hunk 2]
```diff
@@ -34,12 +34,13 @@
...
+TRACTION_QC_URL = f"http://{os.environ.get('LOCALHOST', '127.0.0.1')}:3100/v1/qc_receptions"
 
 EBI_TAXONOMY_URL = "https://www.ebi.ac.uk/ena/taxonomy/rest/tax-id"
 
 # Options: 'json' or 'binary'
 SELECTED_ENCODER_FOR_FEEDBACK_MESSAGE = "binary"
 
 # Validate SSL certificate chain when connecting to https
-CERTIFICATES_VALIDATION_ENABLED = True
+CERTIFICATES_VALIDATION_ENABLED = False
```
[the remark this answers is not in the document]
Comment (Sangeetha-Bheeman) [reply tol_lab_share/config/defaults.py]: Done

File: tol_lab_share/message_properties/definitions/final_nano_drop_280.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    """MessageProperty subclass to manage parsing of a valid Final NanoDrop 260/280 string provided by another
+    MessageProperty.
+    The Final NanoDrop 260/280 has to be a string.
+    Eg: '260'
+    """
+
+    @property
+    def validators(self) -> List[Callable]:
+        """Defines the list of validators"""
+        return [self.check_is_string]
```
[the remark this answers is not in the document]
Comment (Sangeetha-Bheeman) [reply tol_lab_share/message_properties/definitions/final_nano_drop_280.py:15]: Yes, I will change this later.

**Comment by Sangeetha-Bheeman** [comment]
From Steve - Update from meeting 27/07: date_required_by and reason_for_priority are no longer required and can be removed from this schema. priority_level is not a qc_result attribute but should be moved to sample.

**Pull request merged**